oops/backplane/unittester_support.py

```diff
@@ -0,0 +1,503 @@
+
+    # Add path
+    filename = os.path.join(dir, filename)
+    return filename
+
+
+
+#===============================================================================
+def _print(*x, printing=True):
+    """Prints contignent upon verbosity."""
+    if not printing:
+        return
+
+    print(*x)
+
+
+#===============================================================================
+def show_info(bp, title, array, printing=True, saving=False, dir='./',
+                                refdir=None):
+    """Internal method to print summary information and display images as
+    desired.
+    """
+
+    import numbers
+
+    if array is None:
+        return
+
+    if not printing and not saving and refdir==None:
+        return
+
+    _print(title, printing=printing)
+
+
+    # Scalar summary
+    if isinstance(array, numbers.Number):
+        _print('  ', array, printing=printing)
+
+    # Mask summary
+    elif type(array.vals) == bool or \
+            (isinstance(array.vals, np.ndarray) and \
+             array.vals.dtype == np.dtype('bool')):
+        count = np.sum(array.vals)
+        total = np.size(array.vals)
+        percent = int(count / float(total) * 100. + 0.5)
+        _print('  ', (count, total-count),
+            (percent, 100-percent), '(True, False pixels)', printing=printing)
+        minval = 0.
+        maxval = 1.
+
+    # Unmasked backplane summary
+    elif array.mask is False:
+        minval = np.min(array.vals)
+        maxval = np.max(array.vals)
+        if minval == maxval:
+            _print('  ', minval, printing=printing)
+        else:
+            _print('  ', (minval, maxval), '(min, max)', printing=printing)
+
+    # Masked backplane summary
+    else:
+        _print('  ', (array.min().as_builtin(),
+            array.max().as_builtin()), '(masked min, max)', printing=printing)
+        total = np.size(array.mask)
+        masked = np.sum(array.mask)
+        percent = int(masked / float(total) * 100. + 0.5)
+        _print('  ', (masked, total-masked),
+             (percent, 100-percent), '(masked, unmasked pixels)', printing=printing)
+
+        if total == masked:
+            minval = np.min(array.vals)
+            maxval = np.max(array.vals)
+        else:
+            minval = array.min().as_builtin()
+            maxval = array.max().as_builtin()
+
+
+
+    # The rest of the method applies only to arrays
+    if array.shape == ():
+        return
+
+
+    # scale image
+    if minval == maxval:
+        maxval += 1.
+    image = _scale_image(array, minval, maxval)
+
+
+    # Save backplane array if directed
+    if saving:
+        os.makedirs(dir, exist_ok=True)
+
+        filename = _construct_filename(bp, array, title, dir)
+        _save_image(image, filename)
+
+
+    # Compare with reference array if refdir is known
+    if refdir is not None:
+        assert os.path.exists(refdir), \
+            "No reference directory.  Use --no-compare, --no-exercises, or --reference."
```

Comment:
It would be helpful to add the reference directory to the error message. I hit this assert but had no idea what it was looking for.

Reply:
Agreed, I meant to do that.